### PR TITLE
add json functions and generated column document.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

File: op-guide/json_functions_generated_column.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+当我们向表中插入数据时，便可以这样处理那些模式不确定的数据了：
+```sql
+INSERT INTO person (name, address_info) VALUES ("Jhon", '{"city": "Beijing"}');
+```
+就这么简单！直接在 JSON 字段对应的位置上，放一个合法的 JSON 字符串，就可以向表中插入 JSON 了。TiDB 会解析这个文本，然后以一种更加紧凑、易于访问的二进制形式来保存。
+
+当然，你也可以将其他类型的数据用 CAST 转换为 JSON：
+```sql
+INSERT INTO person (name, address_info) VALUES ("Jhon", CAST('{"city": "Beijing"}' AS JSON));
```
Comment (jackysp) [inline op-guide/json_functions_generated_column.md]: "Jhon" should be "John".

File: README.md [hunk 2]
```diff
@@ -21,6 +21,7 @@
 + SQL 兼容及对比
   - [TiDB SQL 文法](sql/README.md)
   - [与 MySQL 兼容性对比](op-guide/mysql-compatibility.md)
+  - [JSON 函数及 Generated Column ](op-guide/json_functions_generated_column.md)
```
Comment (andelf) [inline README.md]: better use `-` instead or underscore(`_`) for filename.
for consistency

File: op-guide/json_functions_generated_column.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+为了在功能上与最新版的MySQL（版本 5.7 及以上）保持同步，同时更好地支持文档存储类业务，我们在最新版本的 TiDB 中加入了 JSON 的支持。用户可以在 TiDB 的表中使用 JSON 类型的字段，同时以生成列（generated column）的方式为 JSON 文档内部的字段建立索引。基于此，用户可以很灵活地处理那些 schema 不确定地业务，同时不必受限于传统文档数据库糟糕的读性能及匮乏的事务支持。
```
Comment (andelf) [inline op-guide/json_functions_generated_column.md]: first line should be meta info, or level-1/level-2 title.

**Comment by hicqu** [comment]
@jackysp @andelf @queenypingcap , PTAL, thanks.

**Comment by jackysp** [comment]
LGTM

**Comment by queenypingcap** [comment]
LGTM

**Pull request merged**